spec/ferto/client_spec.rb

```diff
@@ -38,7 +38,9 @@
         url: 'https://foo.bar/a.jpg',
         callback_type: 'my-callback-mechanism',
         callback_dst: 'http://example.com/downloads/myfile',
-        extra: { product: 1234, actor: 'actor1' }
+        user_agent: "Downloader Agent v1.0",
```

Comment:
These headers are not mandatory for the request to succeed. The only use, I see here, is as an example on the available params that can be used, they are not validated anywhere.

You could remove them from the params of these tests and add another test that they are passed correctly on `post`, checking the logic in https://github.com/skroutz/ferto/pull/10/files#diff-97a5428366001f02b47ea93cedf16f07R170

Reply:
Added specs.